tapestry.js

```diff
@@ -975,39 +992,109 @@ function getBoundedCoord(coord, maxCoord) {
     return Math.max(MAX_RADIUS, Math.min(maxCoord - MAX_RADIUS, coord));
 }
 
-function getChildren(id) {
-    var children = [];
-    var dataLinks = dataset.links;
-    for (var linkId in dataLinks) {
-        var link = dataLinks[linkId];
-        //SEARCH FOR: CURRENT_ID_NODE -> CHILD links
-        if (typeof link.source === 'number' && link.source === id) {
-            children.push(link.target);
-        } else if (typeof link.source === 'object' && link.source.id === id) {
-            children.push(link.target.id);
-        }
+/* Add 'depth' parameter to each node recursively. 
+   The depth is determined by the number of levels from the root each node is. */
+
+function addDepthToNodes(rootId, depth, visited) {
+    visited.push(rootId);
+
+    var depthAt = 0;
+
+    dataset.nodes[findNodeIndex(rootId)].depth = depth;
+    var children = getChildren(rootId, 1);
+
+    var childLevel;
+
+    while (depthAt < children.length) {
+        for (var childId in children) {
+            if (visited.includes(children[childId])) {
+                childLevel = depth;
+                if (dataset.nodes[findNodeIndex(children[childId])].depth > childLevel) {
+                    dataset.nodes[findNodeIndex(children[childId])].depth = childLevel;
+                }
+                else {
+                    depthAt++;
+                }
+            }
+            else {
+                childLevel = depth + 1;
+                dataset.nodes[findNodeIndex(children[childId])].depth = childLevel;
+                visited.push(children[childId]);
 
-        //SEARCH FOR: PARENT -> CURRENT_ID_NODE links;
-        // These should also appear as "children" because they're adjacent to the current node
-        if (typeof link.target === 'number' && link.target === id) {
-            children.push(link.source);
-        } else if (typeof link.target === 'object' && link.target.id === id) {
-            children.push(link.source.id);
+                addDepthToNodes(children[childId], childLevel, visited);
+                depthAt++;
+            }
         }
     }
+    
+}
+
+/* Return the distance between the rootId and its farthest descendant node. */
+
+function findMaxDepth(rootId) {
+    addDepthToNodes(rootId, 0, []);
+    var nodes = dataset.nodes;
```

Comment:
Please rename your variables to better describe what they are supposed to keep. Additionally, you could also add a comment to describe what they each do where you declare them.